src/Compilers/CSharp/Portable/Symbols/Extensions/SourceExtensionImplementationMethodSymbol.cs

```diff
@@ -78,10 +81,38 @@ internal sealed override void AddSynthesizedAttributes(PEModuleBuilder moduleBui
                 }
             }
 
-            base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
             SourceMethodSymbol.AddSynthesizedAttributes(this, moduleBuilder, ref attributes);
         }
 
+        internal override void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<CSharpAttributeData> attributes)
+        {
+            // Copy NotNull/MaybeNull attribute from the property onto the implementation getter
+            if (_originalMethod is SourcePropertyAccessorSymbol { MethodKind: MethodKind.PropertyGet, AssociatedSymbol: SourcePropertySymbolBase extensionProperty })
```

Comment:
>SourcePropertyAccessorSymbol { MethodKind: MethodKind.PropertyGet, AssociatedSymbol: SourcePropertySymbolBase extensionProperty } [](http://example.com/codeflow?start=35&length=129)

I think checks for specific implementation types are unnecessary and undesirable.

Reply:
I was not able to solve this. I'll let you take a look at update and we can discuss

Reply:
> I was not able to solve this. I'll let you take a look at update and we can discuss

With the helper in ```SourcePropertyAccessorSymbol``` that we are using, I think it is appropriate to do a type check for ```SourcePropertyAccessorSymbol``` and that is the only check that we would need before calling the helper.

Reply:
Thanks for the clarification. An instance method does factor better.